### PR TITLE
Updated "has session" check

### PR DESCRIPTION
**Pull request by Richard-Walton**
Fix for: https://github.com/tmuxinator/tmuxinator/issues/402

Also, improved template rendering to eliminate dead code if the session exists.

I'm not well versed in Ruby code - Apologies if I've done something crazy!

### CODE REVIEW

**Comment by Richard-Walton** [comment]
Fix for: https://github.com/tmuxinator/tmuxinator/issues/402

Also, improved template rendering to eliminate dead code if the session exists.

I'm not well versed in Ruby code - Apologies if I've done something crazy!

File: lib/tmuxinator/project.rb [hunk 1]
```diff
@@ -147,6 +147,12 @@ def tmux_command
       yaml["tmux_command"] || "tmux"
     end
 
+    def tmux_has_session?(name)
+      sessions = `#{tmux_command} ls`
+
+      !sessions.match("^#{name}:")
```
Comment (J3RN) [inline lib/tmuxinator/project.rb]: I'm a bit confused as to why there's a `!` here.

The way I read it, if the session exists, `session.match("^#{name}:")` should return a match. `not`ing a match would return `false`, which is returned from the method. Shouldn't the method return `true` if the session exists?

If you were trying to make a boolean from the match object, you can do `!!`. For example:
```ruby
!! "foo" #=> true
```

Comment (Richard-Walton) [inline lib/tmuxinator/project.rb]: Err..., yeah that's just poor logic.  My bad - I was rushing to get this pushed so I could get back to work.

Fixed in next commit.

**Comment by Richard-Walton** [comment]
Anything else need changing?

**Comment by adamstrickland** [comment]
@Richard-Walton thanks for asking...  I would like to see some tests.  I think two tests for `Project#tmux_has_session?` are called for, one for the positive case, one for the negative.  Something along the lines of:

```ruby
  describe "#tmux_has_session?" do
    let(:reply) { "foo: 1 window (created blah blah blah) [500x500] (detached)\nbar: 1 window (created blah blah blah) [500x500] (detached)" }

    before do
      expect(Kernel).to receive(:system).with("tmux ls"){ reply }
    end

    it "should be true when 'tmux ls` has the named session" do
      expect(subject.tmux_has_session?("foo")).to be_true
    end

    it "should be false when 'tmux ls` does not have the named session" do
      expect(subject.tmux_has_session?("quux")).not_to be_true
    end
  end
```

**Comment by Richard-Walton** [comment]
@adamstrickland Ok, added tests.  Again, I'm not a ruby dev so this code may be crazy. 

Related note: The rubocop line length constraint is difficult to work with.  This some of the multiline strings in the tests.

**Pull request merged**